### PR TITLE
Fix documentation links and clippy lints

### PR DESCRIPTION
**Pull request by hrxi**
Fix `clippy::unnecessary_mut_passed` and ignore the false-positive `clippy::redundant_async_block`. Fix broken links in documentation.

### CODE REVIEW

File: client/src/main.rs [hunk 1]
```diff
@@ -85,6 +85,8 @@ async fn main_inner() -> Result<(), Error> {
         use nimiq::extras::rpc_server::initialize_rpc_server;
         let rpc_server = initialize_rpc_server(&client, rpc_config, client.wallet_store())
             .expect("Failed to initialize RPC server");
+        // false positive: https://github.com/rust-lang/rust-clippy/issues/10482
+        #[allow(clippy::redundant_async_block)]
```
Comment (jsdanielh) [inline client/src/main.rs]: The CI is reporting this as an unknown lint :thinking:

File: spammer/src/main.rs [hunk 2]
```diff
@@ -211,6 +211,8 @@ async fn main_inner() -> Result<(), Error> {
         use nimiq::extras::rpc_server::initialize_rpc_server;
         let rpc_server = initialize_rpc_server(&client, rpc_config, client.wallet_store())
             .expect("Failed to initialize RPC server");
+        // false positive: https://github.com/rust-lang/rust-clippy/issues/10482
+        #[allow(clippy::redundant_async_block)]
```
Comment (jsdanielh) [inline spammer/src/main.rs]: The CI is reporting this as an unknown lint :thinking:

File: client/src/main.rs (hunk 1 above)
Comment (hrxi) [reply client/src/main.rs]: Should I do it like the matrix-rust-sdk commit (i.e. `#[allow(unknown_lints, clippy::redundant_async_block)]`) or just drop the `#[allow]` until the CI gets the lint?

https://github.com/matrix-org/matrix-rust-sdk/commit/8d83a996a69ffcfca786dbc343dc9b862a9965df#diff-f4219660dd4687809f8c9f0a5f4f17b5e782ea7e28f5b6e40cccbb67c38e1308R380

Comment (jsdanielh) [reply client/src/main.rs]: Mmm, I'd vote for removing it until the CI gets it because it's a false positive that is probably not in stable yet?

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**